hub/util/cache_chain.py

```diff
@@ -23,15 +25,48 @@ def get_cache_chain(provider_list: List[StorageProvider], size_list: List[int]):
         ProviderListEmptyError: If the provider list is empty.
         ProviderSizeListMismatch: If the len(size_list) + 1 != len(provider_list)
     """
-    if not provider_list:
+    if not storage_list:
         raise ProviderListEmptyError
-    if len(provider_list) <= 1:
-        return provider_list[0]
-    if len(size_list) + 1 != len(provider_list):
+    if len(storage_list) <= 1:
+        return storage_list[0]
+    if len(size_list) + 1 != len(storage_list):
         raise ProviderSizeListMismatch
-    provider_list.reverse()
+    storage_list.reverse()
     size_list.reverse()
-    store = provider_list[0]
-    for size, cache in zip(size_list, provider_list[1:]):
+    store = storage_list[0]
+    for size, cache in zip(size_list, storage_list[1:]):
         store = LRUCache(cache, store, size)
     return store
+
+
+def generate_chain(
+    base_storage: StorageProvider,
+    memory_cache_size: int,
+    local_cache_size: int,
+    path: str,
+):
+    """Internal function to be used by Dataset, to generate a cache_chain using a base_storage and sizes of memory and local caches.
+
+    Args:
+        base_storage (StorageProvider): The underlying actual storage of the Dataset.
+        memory_cache_size (int): The size of the memory cache to be used in MB.
+        local_cache_size (int): The size of the local filesystem cache to be used in MB.
+        path (str): The location of the dataset.
```

Comment:
Can you say a little more about what this param is used for? It's not clear from the outside how this will affect the path for the cache providers